modules/caddyhttp/responsewriter.go

```diff
@@ -258,6 +275,24 @@ type hijackedConn struct {
 	rr *responseRecorder
 }
 
+func (hc *hijackedConn) updateReadSize(n int) {
+	if hc.rr.readSize != nil {
+		*hc.rr.readSize += n
+	}
+}
+
+func (hc *hijackedConn) Read(p []byte) (int, error) {
+	n, err := hc.Conn.Read(p)
+	hc.updateReadSize(n)
+	return n, err
+}
+
+func (hc *hijackedConn) WriteTo(w io.Writer) (int64, error) {
+	n, err := io.Copy(w, hc.Conn)
+	hc.updateReadSize(int(n))
```

Comment:
Interesting, why do we update read size when writing?

Reply:
It's writing to another writer, not actually writing to itself.